java/src/jmri/jmrix/can/cbus/swing/console/CbusConsoleStatsPane.java

```diff
@@ -99,30 +116,86 @@ private void initButtonBorderToolTips(){
+        framesLastSecondField.setBorder(BorderFactory.createTitledBorder(
+                BorderFactory.createEtchedBorder(), Bundle.getMessage("FramesPerSecond")));
+
+        meanFramesPerSecondField.setToolTipText(Bundle.getMessage("AverageFramesPerSecondTip"));
+        meanFramesPerSecondField.setBorder(BorderFactory.createTitledBorder(
+                BorderFactory.createEtchedBorder(), Bundle.getMessage("AverageFramesPerSecond")));
+
+        maxFramesPerSecondField.setToolTipText(Bundle.getMessage("MaxFramesPerSecondTip"));
+        maxFramesPerSecondField.setBorder(BorderFactory.createTitledBorder(
+                BorderFactory.createEtchedBorder(), Bundle.getMessage("MaxFramesPerSecond")));
+
+        statsClearButton.setText(Bundle.getMessage("ButtonReset"));
         statsClearButton.setVisible(true);
-        statsClearButton.setToolTipText(Bundle.getMessage("TooltipClearCounters"));
-    
     }
-    
+
     private void statsClearButtonActionPerformed(java.awt.event.ActionEvent e) {
-        _sent = 0;
-        _rcvd = 0;
-        _events = 0;
-        _dcc = 0;
-        _total = 0;
-        sentCountField.setText("0");
-        rcvdCountField.setText("0");
-        eventsCountField.setText("0");
-        dccCountField.setText("0");
-        totalCountField.setText("0");
+        frameTimes.clear();
+        startTime = System.currentTimeMillis();
+        sentTotal = 0;
+        rcvdTotal = 0;
+        eventTotal = 0;
+        dccTotal = 0;
+        total = 0;
+        maxPerSecondCount = 0;
     }
 
-    // private final static Logger log = LoggerFactory.getLogger(CbusConsoleStatsPane.class);
+    /**
+     * Set up the GUI Update Timer, and start it.
+     */
+    private void startUpdateTimer() {
+        disposed = false;
+        startTime = System.currentTimeMillis();
+        keepAliveTimer = new TimerTask(){
+            @Override
+            public void run () {
+                long currentTime = System.currentTimeMillis();
+                float secsDuration = (currentTime-startTime)/1000f;
+                framesLastSecondField.setText(Integer.toString(
+                    countFramesInLastSecond(currentTime)));
+                maxFramesPerSecondField.setText(Integer.toString(maxPerSecondCount));
+                float average = total / secsDuration;
+                meanFramesPerSecondField.setText(String.format("%.01f", average));
+                totalCountField.setText(Integer.toString(total));
+                rcvdCountField.setText(Integer.toString(rcvdTotal));
+                sentCountField.setText(Integer.toString(sentTotal));
+                eventsCountField.setText(Integer.toString(eventTotal));
+                dccCountField.setText(Integer.toString(dccTotal));
+                statsClearButton.setToolTipText(Bundle.getMessage("ResetButtonLastRestTip",
+                    String.format("%.01f", secsDuration)));
+                if ( !disposed ) {
```

Comment:
I suggest that you change this method so that it returns immediately if `disposed` is true. If the panel is disposed, it might not be good to update the GUI.

Reply:
Sure, updated.